### PR TITLE
PTDM subscribes to relevant events only

### PR DESCRIPTION
**Pull request by jcapona**
#### Main changes

Improve `PTDMSubscribeClient` structure by:
- Only subscribe to the events that were marked as relevant according to the dictionary passed to the `initialise` method. By doing this, overhead is reduced since it's not necessary to filter the received messages once they're received since now when that happens, we're 100% sure it's from an event we're interested in.
- Stop using `zmq.Poller`, since it's not necessary. The `recv_string` method will unblock when a valid message is received.

#### Screenshots (feature, test output, profiling, dev tools etc)

[insert screenshots here]

#### Other notes (e.g. implementation quirks, edge cases, questions / issues)
-

#### Manual testing tips
-

#### Tag anyone who definitely needs to review or help
-

### CODE REVIEW

File: packages/common/pitop/common/ptdm.py [hunk 1]
```diff
@@ -457,35 +459,37 @@ def __thread_method(self):
...
-        if func_arg_no > 1:
-            logger.error(
-                "Invalid callback function - it should receive at most one argument."
-            )
-            return ""
+                callback = self._callback_funcs.get(message.message_id())
+                if callback:
+                    self.invoke_callback_func_if_exists(callback, message.parameters)
 
+    def invoke_callback_func_if_exists(self, callback, params=list()):
```
Comment (angusjfw) [inline packages/common/pitop/common/ptdm.py:470]: i'm not sure but this looks like it would have the same issue as using `[]` as a default, that the same list is actually reused each time. probably not an issue since you don't mutate it, but I still think it's better practice to use None as the default and then have like `params = list() if params is None else params`

https://nikos7am.com/posts/mutable-default-arguments/

Comment (angusjfw) [reply packages/common/pitop/common/ptdm.py:470]: also, not new things but:
- the naming of this function is confusing me - it doesn't check wether the callback exists it just calls it?
- below is `if params == list()` just a weird way to check if the list is empty? usually people check the length?

File: packages/common/pitop/common/ptdm.py [hunk 2]
```diff
@@ -457,35 +459,37 @@ def __thread_method(self):
...
+        func_arg_no = len(signature(callback).parameters)
         if params == list() or func_arg_no == 0:
-            func()
+            callback()
         else:
-            func(params)
+            callback(params)
```
Comment (angusjfw) [inline packages/common/pitop/common/ptdm.py:475]: these lines aren't covered by tests - maybe you could quickly add one where the callback is actually called?

**Review by angusjfw — COMMENTED** [review]
the message filtering thing seems to make sense and be correct to me, but it seems like you have removed parts of the test that actually check the callbacks work
